### ISSUE TITLE
Frontend v1.0.1 is not available as an image 

### ISSUE THREAD

**Issue by StanfordLin** [consul-up/birdwatcher#1]
Was able to pull backend 1.0.1 but not the frontend 1.0.1 version, failing when trying to start on M1 macbook

`kubectl logs frontend` gives me the following error
exec /bin/frontend: exec format error

temp fix: 
download source code
`cd frontend`
docker build 

docker image inspect, ensure it is arm64

docker push to my own registry

specify it in my manifests folder frontend deployment yaml

success :)

**Comment by lkysow** [comment]
Hi @StanfordLin sorry about that. I need to rebuild that image for multi-arch.

**Comment by qd3v** [comment]
For me, unless I miss something, all related examples in book are broken:

```sh
docker run --rm ghcr.io/consul-up/birdwatcher-frontend:1.0.0
docker: Error response from daemon: Head "https://ghcr.io/v2/consul-up/birdwatcher-frontend/manifests/1.0.0": denied: denied.
See 'docker run --help'.

docker run --rm ghcr.io/consul-up/birdwatcher-backend:1.0.0
docker: Error response from daemon: Head "https://ghcr.io/v2/consul-up/birdwatcher-backend/manifests/1.0.0": denied: denied.
See 'docker run --help'.
```

Also, it's probably worth using GHA to build, push, and publish containers on this repo's Packages page?

**Comment by lkysow** [comment]
Oh no, sorry I'll look into it as soon as I can. Apologies.

**Comment by lkysow** [comment]
Hi @qd3v this was working for me today:
```
docker run --rm ghcr.io/consul-up/birdwatcher-frontend:1.0.0
Unable to find image 'ghcr.io/consul-up/birdwatcher-frontend:1.0.0' locally
1.0.0: Pulling from consul-up/birdwatcher-frontend
0ccc01f8926d: Download complete
59bf1c3509f3: Download complete
0f50d6197eff: Download complete
7d6924f8acff: Download complete
Digest: sha256:5925990c7ae586c2c4349383d329f723ddd7b44e0bd2c393cf5ee1c6228ebebb
Status: Downloaded newer image for ghcr.io/consul-up/birdwatcher-frontend:1.0.0
2025/02/09 17:32:32 Starting server listen_addr="0.0.0.0:6060"
```

Can you check that your login to GHCR is working? Can you pull other GHCR images? If the image couldn't be found you'd get a different error: `not found.` so I'm thinking maybe it's to do with your authentication.

**Comment by qd3v** [comment]
@lkysow yep, I was thinking this `denied` error means the same as `not found`, like github does for private repos if you don't have access to them.
Anyways that `denied` error is confusing, because in order to download public images you don't need to login to ghcr/docker. 
In my case, I think, there were stale ghcr creds left in docker machine (macos), and that can be fixed with `docker logout ghcr.io`.
Sorry for bothering, and thank you!
